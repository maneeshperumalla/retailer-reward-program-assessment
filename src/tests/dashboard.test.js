import React from 'react';
import Dashboard from '../containers/dashboard';
import getCustomersList from '../services/getCustomersApi';
import fetchMock from 'jest-fetch-mock';
import renderer from 'react-test-renderer';
import { props } from './props';



describe('render dashboard', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
  });
  test('should be able to fetch api response and render table', () => {
    fetchMock.mockResponseOnce(JSON.stringify(props));
    const onResponse = jest.fn();
    const onError = jest.fn();

    return getCustomersList()
      .then(onResponse)
      .catch(onError)
      .finally(() => {
        expect(onResponse).toHaveBeenCalled();
        expect(onError).not.toHaveBeenCalled();
        expect(onResponse.mock.calls[0][0]).toEqual(props);
      });
  });
});

describe('render dashboard container', () => {
  it('should render container and match shanpshot', () => {
    const container = renderer.create(<Dashboard />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
