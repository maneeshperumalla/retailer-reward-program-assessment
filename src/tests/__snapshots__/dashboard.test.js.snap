// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dashboard container should render container and match shanpshot 1`] = `
<div
  className="container"
>
  <header
    className="container-header"
  >
    Retailer Dashboard
  </header>
  <div
    className="container-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead
          className="header-thead"
        >
          <tr
            className="header-tr"
          >
            <td>
              Customer ID
            </td>
            <td>
              Name
            </td>
            <td>
              Total Orders
            </td>
            <td>
              Total Amount
            </td>
            <td>
              Total Rewards
            </td>
            <td>
              Action
            </td>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
